From 4ea9b95b11f03dd70b7f9f0931aca2ebec1617b6 Mon Sep 17 00:00:00 2001
From: minaripenguin <minaripenguin@users.noreply.github.com>
Date: Sun, 25 Dec 2022 01:29:33 +0800
Subject: [PATCH 12/12] base: Do not reset keyguard going away state

- Fixes the flicker on turning of the screen when udfps is enabled

Co-authored-by: Blaster4385 <venkateshchaturvedi12@gmail.com>
Change-Id: I9960b4e2f3ad5fbb04969224bd8c2eec1cd1f1ee
Signed-off-by: minaripenguin <minaripenguin@users.noreply.github.com>
Signed-off-by: penglezos <panagiotisegl@gmail.com>
---
 .../android/systemui/keyguard/KeyguardViewMediator.java    | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/packages/SystemUI/src/com/android/systemui/keyguard/KeyguardViewMediator.java b/packages/SystemUI/src/com/android/systemui/keyguard/KeyguardViewMediator.java
index 6034e92be873..8db1d2059c29 100644
--- a/packages/SystemUI/src/com/android/systemui/keyguard/KeyguardViewMediator.java
+++ b/packages/SystemUI/src/com/android/systemui/keyguard/KeyguardViewMediator.java
@@ -118,6 +118,7 @@ import com.android.internal.policy.IKeyguardExitCallback;
 import com.android.internal.policy.IKeyguardStateCallback;
 import com.android.internal.policy.ScreenDecorationsUtils;
 import com.android.internal.statusbar.IStatusBarService;
+import com.android.internal.util.ArrayUtils;
 import com.android.internal.util.LatencyTracker;
 import com.android.internal.widget.LockPatternUtils;
 import com.android.keyguard.KeyguardConstants;
@@ -1666,7 +1667,11 @@ public class KeyguardViewMediator implements CoreStartable, Dumpable,
         // explicitly DO NOT want to call
         // mKeyguardViewControllerLazy.get().setKeyguardGoingAwayState(false)
         // here, since that will mess with the device lock state.
-        mUpdateMonitor.dispatchKeyguardGoingAway(false);
+        int[] udfpsProps = mContext.getResources().getIntArray(
+                com.android.internal.R.array.config_udfps_sensor_props);
+        if (ArrayUtils.isEmpty(udfpsProps)) {
+            mUpdateMonitor.dispatchKeyguardGoingAway(false);
+        }
 
         notifyStartedGoingToSleep();
     }
-- 
2.45.2

